### ISSUE TITLE
Replacing regions with the active region interferes with ARNs specifying non-active regions

### ISSUE THREAD

**Issue by brettdh** [svdgraaf/serverless-pseudo-parameters#2]
I have a CloudFormation template that creates permissions for IAM users that are scoped to a particular region. For example (representative, but not tested):
```yaml
service: foo-testuser

provider:
  name: aws
  region: us-east-1

custom:
  userRegion: us-east-2

plugins:
  - serverless-pseudo-parameters

resources:
  Resources:
    TestUser:
      Type: AWS::IAM::User
      Properties:
        UserName: testuser
        Policies:
          - PolicyName: allow-cloudformation
            PolicyDocument:
              Version: "2012-10-17"
              Statement:
                - Effect: "Allow"
                  Action:
                    - "cloudformation:DescribeStacks"
                    - "cloudformation:DescribeStackEvents"
                    - "cloudformation:DescribeStackResource*"
                    - "cloudformation:GetTemplate"
                    - "cloudformation:CreateStack"
                    - "cloudformation:UpdateStack"
                    # not allowed to delete stack; that must be done manually by admin
                  Resource:
                    "arn:aws:cloudformation:${self:custom.userRegion}:#{AWS::AccountId}:stack/foo-testuser-dev/*"
          - PolicyName: allow-cloudformation-global
            PolicyDocument:
              Version: "2012-10-17"
              Statement:
                - Effect: "Allow"
                  Action:
                    - "cloudformation:ValidateTemplate"
                  Resource:
                    "*"
          - PolicyName: allow-apigateway
            PolicyDocument:
              Version: "2012-10-17"
              Statement:
                - Effect: "Allow"
                  Action:
                    - "apigateway:*"
                  Resource:
                    "arn:aws:apigateway:${self:custom.userRegion}::/restapis"
          - PolicyName: allow-ssm-access
            PolicyDocument:
              Version: "2012-10-17"
              Statement:
                - Effect: "Allow"
                  Action:
                    - "ssm:DescribeParameters"
                    - "ssm:GetParameter"
                    - "ssm:GetParameters"
                    - "ssm:List*"
                  Resource:
                    "arn:aws:ssm:${self:custom.userRegion}:#{AWS::AccountId}:parameter/testuser/dev/*"
```
However, `serverless-pseudo-parameters` overwrites these regions with `#{AWS::Region}`:

https://github.com/svdgraaf/serverless-pseudo-parameters/blob/master/lib/index.js#L65

which prevents me from specifying a region other than the one I'm using with `serverless deploy --region <region>`. This breaks a valid use case: IAM users are global, but their permissions can refer to region-specific resources.

Why does this region overwriting exist in the first place? Can its purpose be accomplished in a way that avoids this problem, or can this overwriting at least be configurable?

**Comment by bamapookie** [comment]
You can ignore region replacement using a parameter.  See the end of the readme for details.

**Comment by svdgraaf** [comment]
Thanks @bamapookie, closing

**Comment by brettdh** [comment]
I see that this has been implemented as of #3. Thanks!
